src/main/java/org/truffleruby/core/CoreLibrary.java

```diff
@@ -1030,7 +1023,8 @@ public void initializeEncodingManager() {
             getContext().getEncodingManager().setDefaultExternalEncoding(getContext().getEncodingManager().getLocaleEncoding());
         }
 
-        final String internalEncodingName = getContext().getOptions().INTERNAL_ENCODING;
+        String internalEncodingName = getContext().getOptions().INTERNAL_ENCODING;
+        internalEncodingName = internalEncodingName.isEmpty() ? null : internalEncodingName;
         if (internalEncodingName != null && !internalEncodingName.isEmpty()) {
```

Comment:
The 2 lines above can all be simplified to `if !internalEncodingName.isEmpty()`.